unitTests/ui/QuerySuggestPreviewTest.ts

```diff
@@ -106,6 +111,32 @@ export function QuerySuggestPreviewTest() {
       done();
     });
 
+    it('on select, logs analytics', async done => {
+      setupQuerySuggestPreview();
+      const suggestionText = 'abcdef';
+      const previews = await triggerPopulateSearchResultPreviewsAndPassTime(suggestionText);
+      previews[0].onSelect();
+      const spy = test.cmp.usageAnalytics.logCustomEvent as jasmine.Spy;
+      const [actionCause, meta, element] = <[
```

Comment:
Rather than destructuring and adding types, we could be more concise by checking the args directly. This also makes sure we are checking the entire `metadata` object rather than property by property.

I also think we can combine the tests into one in this specific case. e.g.

```
const expectedMeta = {
     suggestion: suggestionText
}

const spy = test.cmp.usageAnalytics.logCustomEvent;

expect(spy).toHaveBeenCalledTimes(1);
expect(spy).toHaveBeenCalledWith(
     analyticsActionCauseList.clickQuerySuggestPreview,
     expectedMeta,
     previews[0].element
);
```

Reply:
Alright, I'll simplify and combine those tests.